ibtk/src/math/HierarchyMathOps.cpp

```diff
@@ -2053,20 +2062,40 @@ HierarchyMathOps::interp(const int dst_idx,
 
             d_patch_math_ops.interp(dst_data, src_data, patch);
 
-            if ((ln > d_coarsest_ln) && src_cf_bdry_synch)
+            if (ln > d_coarsest_ln)
             {
-                Pointer<OutersideData<NDIM, double> > os_data = patch->getPatchData(d_os_idx);
-                os_data->copy(*src_data);
+                if (src_cf_bdry_synch)
+                {
+                    Pointer<OutersideData<NDIM, double> > os_data = patch->getPatchData(d_os_idx);
+                    os_data->copy(*src_data);
+                }
+                if (dst_cf_bdry_synch)
+                {
+                    Pointer<OuternodeData<NDIM, double> > on_data = patch->getPatchData(d_on_idx);
+                    on_data->copy(*dst_data);
+                }
             }
         }
 
         // Synchronize the coarse-fine interface and deallocate temporary data.
-        if ((ln > d_coarsest_ln) && src_cf_bdry_synch)
+        if (ln > d_coarsest_ln && src_cf_bdry_synch)
         {
             xeqScheduleOutersideRestriction(src_idx, d_os_idx, ln - 1);
             level->deallocatePatchData(d_os_idx);
         }
+        if (ln + 1 <= d_finest_ln && dst_cf_bdry_synch)
+        {
+            xeqScheduleOuternodeRestriction(dst_idx, d_on_idx, ln);
+            d_hierarchy->getPatchLevel(ln + 1)->deallocatePatchData(d_on_idx);
+        }
     }
+
+    // fixup data at hanging nodes.
+    if (dst_cf_bdry_synch)
```

Comment:
Should this also be conditional on whether the `Variable` is configured for the coarse value to represent the solution on coarse-fine interfaces?

Reply:
We also should go back and use this with the nodal vorticity operator.

Reply:
Hm. We don't really support the "coarse values represent the solution case" because IIRC those coarsening operators haven't been written yet, so I'd prefer to ignore that check.